File: 5Days-SoftwareArchitecture/typesript/Day2/Main_seq/Main_seq.js
// Concrete component
var CustomerManager = /** @class */ (function () {
    function CustomerManager() {
    }
    CustomerManager.prototype.processTransaction = function (amount) {
        console.log("Processing customer transaction: \u20B9".concat(amount));
    };
    return CustomerManager;
}());
// Usage
var service = new CustomerManager();
service.processTransaction(500);

File: 5Days-SoftwareArchitecture/Java/Day2/Main_seq/Main_seq.java
// Abstract component (stable)
interface BankService {   // A = 1
    void processTransaction(double amount);
}

// Concrete component (unstable)
class CustomerManager implements BankService { // A = 0
    public void processTransaction(double amount) {
        System.out.println("Processing customer transaction: ₹" + amount);
    }
}

// Client code
public class Main_seq {
    public static void main(String[] args) {
        BankService service = new CustomerManager(); // depends on interface (good)
        service.processTransaction(500);
    }
}
